app/herdbook.py

```diff
@@ -238,33 +241,82 @@ def individual(i_id):
     """
     user_id = session.get("user_id", None)
     ind = da.get_individual(i_id, user_id)
+
     if ind:
-        ind["inbreeding"] = "%.2f" % (get_inbreeding(ind['id']) * 100)
+        ind["inbreeding"] = "%.2f" % (get_inbreeding(ind['id'], ind['genebank_id']) * 100)
     return jsonify(ind)
 
-def get_inbreeding(i_id):
+
+def get_inbreeding(i_id, g_id):
     """
     Returns  the inbreeding coefficient of the individual given by `i_id`.
     """
     id = str(i_id)
-    coefficients = load_inbreeding()
+    coefficients = load_inbreeding(g_id)
     if id in coefficients:
         return coefficients[id]
     return None
 
 
-@APP.route("/api/inbreeding/")
-def all_inbreeding():
-    coefficients = load_inbreeding()
-    return jsonify({"coefficients": coefficients})
+@APP.route("/api/<int:g_id>/inbreeding/")
+def inbreeding(g_id):
+    id = str(g_id)
+    inb_coeffcient = load_inbreeding(id)
+    return jsonify(inb_coeffcient)
+
+
+def load_inbreeding(g_id):
+    response = requests.get('http://{}:{}/inbreeding/{}'.format(settings.rapi.host, settings.rapi.port, g_id),
+                             params={})
+
+    if response.status_code == 200:
+        return csvparser.parse_csv(response)
+
+    else:
+        APP.logger.error("Could not fetch inbreeding data.")
+        APP.logger.error("Error {}".format(response))
+        return {}
+
+
+@APP.route("/api/<int:g_id>/kinship/")
+def kinship(g_id):
+    id = str(g_id)
+    kinship = load_kinship(id)
+    return jsonify(kinship)
+
 
+def load_kinship(g_id):
+    response = requests.get('http://{}:{}/kinship/{}'.format(settings.rapi.host, settings.rapi.port, g_id),
+                             params={})
 
-@APP.before_first_request
-@cache.cached(timeout=3600, key_prefix="all_inbreeding")
-def load_inbreeding():
-    collections = ibc.get_pedigree_collections()
-    coefficients = ibc.calculate_inbreeding(collections)
-    return coefficients
+    if response.status_code == 200:
+        return csvparser.parse_kinship(response)
+
+    else:
+        APP.logger.error("Could not fetch kinship data.")
+        APP.logger.error("Error {}".format(response))
+        return {}
+
+
+@APP.route("/api/<int:g_id>/kinship/")
+def mean_kinship(g_id):
+    id = str(g_id)
+    mean_kinship = load_mean_kinship(id)
+    return jsonify(mean_kinship)
+
+
+@APP.route("/api/<int:g_id>/meankinship/")
+def load_mean_kinship(g_id):
```

Comment:
Should probably use individual number here instead of database id! :)

Reply:
this is the Genebank ID since we need to do the calculation for one whole pedigree at the time